FBShimmering/FBShimmeringLayer.m

```diff
@@ -244,6 +244,22 @@ - (void)setShimmeringOpacity:(CGFloat)shimmeringOpacity
   }
 }
 
+- (CFTimeInterval)shimmeringBeginTime
+{
+  return [_maskLayer animationForKey:kFBShimmerSlideAnimationKey].beginTime;
+}
+
+- (void)resumeShimmeringWithBeginTime:(CFTimeInterval)beginTime
+{
+  CAAnimation *slideAnimation = [[_maskLayer animationForKey:kFBShimmerSlideAnimationKey] mutableCopy];
+  if (nil == slideAnimation) {
+    return;
+  }
+
+  slideAnimation.beginTime = beginTime;
+  [_maskLayer addAnimation:slideAnimation forKey:kFBShimmerSlideAnimationKey];
```

Comment:
Rather than adjusting this here, you could make the animation creation use beginTime if it's nonzero. Then you'd just have to set the property before you call -startShimmering.